### PR TITLE
Market default settings and market update options

### PR DESCRIPTION
**Pull request by KristianMSchmidt**

### CODE REVIEW

**Comment by KristianMSchmidt** [comment]
New stuff in this pull requests: 
- I've made a json file with some default market settings. The parameter values are a bit random at this point. 
- The market model now have to new field that are customizable: "product_name" and "initial_balance". The trader model no longer has an initial_balance field. 
- When creating a market, users can choose between the different default markets settings 
- I've removed default market values from model definition (market settings are thus only defined in the json-file).
- Users can edit markets they have created 

I will probably keep adding more commits untill the pr is merged...

**Review by JensKanstrupLarsen — APPROVED** [review]

File: market/views.py [hunk 1]
```diff
@@ -49,7 +71,12 @@ def create(request):
             return redirect(reverse('market:monitor', args=(new_market.market_id,)))
     elif request.method == 'GET':
         form = MarketForm()
-    return render(request, 'market/create.html', {'form': form})
+    
+    # code smell: path to should probably not be hard coded
```
Comment (dybber) [inline market/views.py:75]: Should perhaps also have some error handling, if the file can't be read or parsed, defaulting to an empty list of default markets.

**Comment by dybber** [comment]
I didn't look through the HTML/template changes and the tests, I don't think I have many comments to the code, but I have some UI comments. It's a bit hard to explain it in text, so I will perhaps wait and talk with you about it tomorrow at the office.

But the main gist of it, is that the teacher should select between more descriptive scenarios, that is, the list should contain "Scenario names", and not name of the goods on the market. A list could then be sort of:

Predefined scenarios:
----------------------
a. Start here - competion w. equal production costs (10 rounds)
b. some other descriptive name (15 rounds)
c. ...

What the descriptions are, comes down to Niels.


Also, it would be good if there was a "Create market" button below the list of scenarios, so you didn't have to scroll all the way down to create the market...

**Comment by KristianMSchmidt** [comment]
I will probably not be at the office before Monday next week. But your comment makes sense, and it shouldn't take long to implement those change. The tests are getting annoying when changing the models. For our next Django project we should use a tool such as FactoryBoy to automate object creation in the test cases.

**Comment by KristianMSchmidt** [comment]
I've tried to make the suggested changes - it's better now but should probably be changed again later, when the scenarios and more features are ready.

File: market/views.py (hunk 1 above)
Comment (KristianMSchmidt) [reply market/views.py]: I will change the format of the file containing the default market settings from .json to .py. Then we don't have to worry about reading or parsing the content

Comment (dybber) [reply market/views.py]: Good idea

**Comment by dybber** [comment]
> I will probably not be at the office before Monday next week. But your comment makes sense, and it shouldn't take long to implement those changes. 

No problem, and the UI changes looks good. Let me know if you're satisfied with this branch, or merge it yourself. I don't have any further comments.

> The tests are getting annoying when changing the models. For our next Django project we should use a tool such as FactoryBoy to automate object creation in the test cases.

Sounds like a good plan. I've previously used it for another Django project I was involved in two years ago, but forgot about it (sorry). It's been refactored a bit since I wrote the first test cases, but there's perhaps some inspiration to get from: https://github.com/CodingPirates/forenings_medlemmer/tree/master/members/tests

**Comment by dybber** [comment]
Just talked with Christian, and he also really likes the new layout with scenarios. He had one comment, which is that the form might not be necessary when the user selects a predefined scenario. I'm not completely certain, so let's leave it as is for now, and fix some other issues

**Comment by dybber** [comment]
👍

**Comment by dybber** [comment]
New version should be online on markedsspillet.dataekspeditioner.dk now. Haven't had time to test it yet. We could send an email to Niels with some questions to get some quick feedback?

**Pull request merged**